src/modules/be-water-smart/be-water-smart.component.ts

```diff
@@ -0,0 +1,686 @@
+      next: response => {
+        if (response && response.hasOwnProperty("message")) {
+          this.models.push(tmp[0]);
+          alert("Model not found");
+        }
+      },
+      error: error => {
+        console.log(error);
+      },
+      complete: () => {
+        this.isDeleting = false;
+      },
+    });
+  }
+
+  // ---------- Forecast Creation -----------
+
+  /**
+   * create forecast by receiving data from api and displaying it in a graph
+   * @returns if a value for the api request is missing
+   */
+  getForecast(): void {
+    if (!this.selectedModel) {
+      alert("No model chosen");
+      return;
+    }
+
+    let vMeterId = this.selectedModel.refMeter;
+    let algId = this.selectedModel.algorithm;
+
+    this.bwsService.getCreateForecast(vMeterId, algId).subscribe({
+      next: response => {
+        if (response.hasOwnProperty("msg")) {
+          console.log(response);
+        } else {
+          let predValues = response.map(item => item.numValue);
+
+          let date = response[0].datePredicted;
+
+          let label =
+            this.transformString(vMeterId, this.prefixes[0]) +
+            " " +
+            algId +
+            " " +
+            this.transformDate(date);
+
+          this.addGraphToChart(predValues, label);
+        }
+      },
+      error: error => {
+        console.log(error);
+      },
+      complete: () => {
+        this.selectedModel = undefined;
+      },
+    });
+  }
+
+  /**
+   * Function to add new lines dynamically to the graph
+   * @param label new data label
+   * @param dataPoints the new prediction values
+   * @param borderColor color to use
+   */
+  addGraphToChart(dataPoints: number[], label: string): void {
+    // Create a new dataset
+    const newDataset: ChartDataset<"line"> = {
+      label: label,
+      data: dataPoints,
+      borderColor: this.generateRandomColor(),
+      fill: false,
+    };
+
+    // Add the new dataset to the existing chart data
+    this.chartData.datasets.push(newDataset);
+
+    // Update the chart to reflect the changes
+    if (this.chart) {
+      this.chart.update();
+    }
+  }
+
+  /**
+   * generate a random color from the color wheel
+   * @returns random color code as string
+   */
+  generateRandomColor(): string {
+    const r = Math.floor(Math.random() * 256); // Random red value (0-255)
+    const g = Math.floor(Math.random() * 256); // Random green value (0-255)
+    const b = Math.floor(Math.random() * 256); // Random blue value (0-255)
+
+    return `rgb(${r}, ${g}, ${b})`; // Return the color in rgb() format
+  }
+
+  // ---------- Utility Functions ----------
+
+  /**
+   * pipe implementation to clean the ids from the smart meters
+   * @param value the whole string to change
+   * @param removable the string to remove
+   * @returns the tidied up string
+   */
+  transformString(value: string, removable: string): string {
+    let t = new TransformStringPipe();
```

Comment:
Never use pipes in components like this, they are used in templates, if they provide a function, we should talk about statically exporting it.